debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/rest/DebeziumMongoDbConnectRestExtension.java

```diff
@@ -27,7 +27,7 @@ public class DebeziumMongoDbConnectRestExtension implements ConnectRestExtension
 
     @Override
     public void register(ConnectRestExtensionContext restPluginContext) {
-        restPluginContext.configurable().register(new DebeziumMongoDbConnectorResource());
+        restPluginContext.configurable().register(new DebeziumMongoDbConnectorResource(restPluginContext.clusterState()));
```

Comment:
Wouldn't setter injection into a MetricsResource interface's default method make more sense? The base code is almost everywhere the same
```
        Map<String, String> connectorConfig = connectClusterState.connectorConfig(connectorName);
        String serverName = connectorConfig.get("topic.prefix");
        String tasksMax = connectorConfig.get("tasks.max");
```
That can be done inside the MetricsResource code in a default method, no?
@jpechane wdyt?

Reply:
Yes, I agree if it is possible.